src/ProjectNeon/Assets/State/Templates/DeckBuilderState.cs

```diff
@@ -8,19 +9,43 @@ public class DeckBuilderState : ScriptableObject
     [SerializeField] private PartyDecks decks;
     [SerializeField] private Hero currentHero;
     [SerializeField] private GameEvent onCurrentDeckChanged;
-    
+    [SerializeField] private Library library;
+
     public GameEvent OnCurrentDeckChanged => onCurrentDeckChanged;
 
-    public Hero CurrentHero => currentHero;
-    
+    public Hero CurrentHero
+    {
+        get => currentHero;
+        set => currentHero = value;
+    }
+
     private Deck current;
     public Deck Current()
     {
         return this.decks.Decks[0];
     }
 
+    public Deck currentHeroDeck()
```

Comment:
I still don't understand this method.

Why is this returning a type of `Deck`?

It looks like the intent is to get all possible cards that could be in the deck. So, is this supposed to be named `GetPossibleDeckCardsInCollection`, which should return a `List<Card>`?

Not all collections of cards are `Decks`.